### DISCUSSION TITLE
NextJS with Headless Wordpress SEO Meta issues

### DISCUSSION THREAD

**Discussion by amandineeap** [vercel/next.js#16392] [Help]
Hi,

I am having issues getting crawlers to register the `<meta>` tags that I am parsing into the `<head>` of my NextJS application, both on pages and dynamic pages.

I am fetching the Wordpress data with `getServerSideProps` on the relevant pages and using Yoast plugin on Wordpress to handle the SEO tags, which gives me an json containing an object that I then parse into html tags to insert into the `<head>`.

Those tags are parsed correctly but nothing gets registered by crawlers on production environment, which is linked to a live domain.

I have tried to use the `getStaticProps` with `revalidate` but got the 
`Error: The `revalidate` property is not yet available for general use.
To try the experimental implementation, please use `unstable_revalidate` instead.`

The site is deployed on Vercel and ideally I want the Wordpress content to be 'refreshed' on each page load dynamically as it is likely to change and would want the changes to reflect immediately.

Not sure what else to try so any suggestions welcome ?

**Comment by isggwp** [comment]
you can build custom meta on head Body using getserversideprops and passing to component.

**Comment by francisrod01** [comment]
@amandineeap According to your error message, please try to update the `next` to the latest version first.

Here's an official example: https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration
You can also do that by Next examples:

A statically generated blog example using Next.js and WordPress
https://github.com/vercel/next.js/tree/canary/examples/cms-wordpress
